subt/doc/virtual-howto.md

```diff
@@ -56,50 +55,59 @@ It is necessary to kill & start again both sim and bridge docker for each run. N
 solution not covered by this HOWTO.
 
 
-## Build image for local testing
+## Working with the main "robotika" image
+
+### Build
 ```
-cd subt/docker
-./build.bash sln_bash
-./run.bash sln_bash
+./subt/docker/build.bash robotika
 ```
 
-Now you are in the docker so you can update current OSGAR:
-```
-cd osgar
-git pull
-cd ..
-osgar/subt/script/sync.sh
+### Run locally
+
+```commandline
+./subt/docker/run.bash robotika
 ```
-And run robotika solution code:
+which by default runs `./osgar/subt/docker/robotika/run_solution.bash` inside the container.
+To get a shell inside the docker instead, run
+```commandline
+./subt/docker/run.bash robotika bash
 ```
-osgar/subt/docker/robotika/run_solution.bash
+and you can call `run_solution.bash` when you are ready.
+
+A copy of `osgar` directory from the time of the build of the image is located
+at `/home/developer/subt_solution/osgar`. For local development it is advantageous
+to mount your `osgar` directory from the host over this directory in the container.
+
+```commandline
+DEVEL=1 ./subt/docker/run.bash robotika bash
 ```
 
+When you do so, you can edit the files as you are used to and also log files
+from the runs are saved to the host and not to the container. To rebuild the ROS
+nodes call `./osgar/subt/script/sync.sh`, to run the solution call 
+`./osgar/subt/docker/robotika/run_solution.bash`.
+
 At this moment you should see waiting for ROS master, debug count outputs of received messages
 (similarly as in unittest) and Python3 outputs of robot navigating towards the gate. The exploration reports
 number and type of received messages by OSGAR.
 
 There is a logfile available when the robot finishes. It is in the current directory with name zmq*.log
-It is necessary to upload the logfile (for example via scp) to the base machine for further analysis, otherwise
-it will be lost with the termination of the docker.
-
-For update it is necessary to repeat all steps above within the docker. You can skip `sync.sh` if there are only
-Python3 code changes. It is necessary to restart the docker to reset it to clean state.
+In case you didn't mount volume over the `osgar` directory, 
+it is necessary to upload the logfile (for example via scp) to the host for further analysis, otherwise
```

Comment:
I find "docker cp" more natural to copy into and out of docker. Plus, scp tends to be slow and our logs are large, so rsync is a better choice.

Reply:
Thanks. I never knew "docker cp" existed.

Reply:
I have replaced `scp` with `docker cp`.